tests/manager/sqlalchemy/model/test_work.py

```diff
@@ -1773,6 +1774,113 @@ def test_active_license_pool_filters_by_library(
         # Without library scope, open_access pool1 should win
         assert work.active_license_pool() == pool1
 
+    def test_active_license_pool_priority_order(self, db: DatabaseTransactionFixture):
+        """
+        Test that active_license_pool correctly prioritizes pools by removing them one at a time.
+        """
+        work = db.work(with_license_pool=True)
+        [pool1] = work.license_pools
+
+        # Pool 1: Regular metered with low availability
+        pool1.type = LicensePoolType.METERED
+        pool1.open_access = False
+        pool1.licenses_owned = 10
+        pool1.licenses_available = 2
+
+        # Create additional pools with different priority levels
```

Comment:
Minor: Might make sense to move the comment on l. 1790 up above Pool 1 and change to something like: "Create multiple pools with different priority levels."